src/test/java/com/jeovan/gymcrmsystem/services/TraineeServiceImplTest.java

```diff
@@ -22,42 +27,65 @@
 
 
 class TraineeServiceImplTest {
-/*
+
     private static final UUID uuid1 = UUID.fromString("983ffb47-5118-40df-b38b-a6604b44555d");
     private static final UUID uuid2 = UUID.fromString("5ab5de9f-dcc2-4677-9457-c63686236f97");
-    private static final Trainee trainee1 = new Trainee(uuid2, Date.from(Instant.now()), "Street 1st #2-34", uuid1);
+    private static final String firstName = "Juan";
+    private static final String lastName = "Perez";
+    private static final String username = "juan.perez";
+    private static final String password = "asdfghjklo";
+    private static final boolean isActive = true;
+    private static final User user = new User(uuid2,firstName, lastName, username, password, isActive);
+
+    private static final Trainee traineeForCreation = new Trainee();
+    private static final User userForCreation = new User();
+    private static final Trainee trainee1 = new Trainee(uuid2, Date.from(Instant.now()), "Street 1st #2-34", user, new ArrayList<>(), new ArrayList<>());
 
     public Map<String, Map<UUID, ? extends SimpleInterface>> storage = new HashMap<>();
 
-    InMemoryStorage inMemoryStorage = mock(InMemoryStorage.class);
+    @Mock
+    private CredentialGeneratorService credentialGeneratorService;
+    @Mock
+    private TrainerService trainerService;
+    @Mock
+    private PasswordEncoder passwordEncoder;
+    @Mock
+    private TraineeValidator traineeValidator;
+    @Mock
+    private TraineeBuilder traineeBuilder;
     @Mock
     TraineeDao mockDao;
     @InjectMocks
-    TraineeServiceImpl service = new TraineeServiceImpl();
+    TraineeServiceImpl service;
 
     @BeforeEach
     void setUp() {
         MockitoAnnotations.openMocks(this);
-        service.setTraineeService();
+        userForCreation.setFirstName(firstName);
+        userForCreation.setLastName(lastName);
+        userForCreation.setIsActive(true);
+        traineeForCreation.setAddress("Street 1st #2-34");
+        traineeForCreation.setUser(userForCreation);
     }
 
     @AfterEach
     void tearDown() {
     }
 
     @Test
-    void testGetAll() {
-        Map<UUID, Trainee> expectedTrainees = Map.of(uuid1, new Trainee(), uuid2, new Trainee());
-        when(mockDao.getAll(any(String.class))).thenReturn(expectedTrainees);
-        assertEquals(expectedTrainees, service.getAll());
+    void shouldReturnANewTraineeWithUsernameAndPassword() {
```

Comment:
Like most software practices we have some convection when we write unit testing, please take a look at this article: https://dzone.com/articles/7-popular-unit-test-naming and pick up the best convection to you.

Reply:
This is really useful, I will marry with one convention and I'll try to be consistent.